x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/diff/calculation/algorithms/multi_line_string_diff_algorithm.ts

```diff
@@ -34,77 +34,91 @@ export const multiLineStringDiffAlgorithm = (
   const diffOutcome = determineDiffOutcome(baseVersion, currentVersion, targetVersion);
   const valueCanUpdate = determineIfValueCanUpdate(diffOutcome);
 
-  const { mergeOutcome, mergedVersion } = mergeVersions({
+  const hasBaseVersion = baseVersion !== MissingVersion;
+
+  const { conflict, mergedVersion } = mergeVersions({
+    hasBaseVersion,
     baseVersion,
     currentVersion,
     targetVersion,
     diffOutcome,
   });
 
   return {
-    base_version: baseVersion,
+    has_base_version: hasBaseVersion,
+    base_version: hasBaseVersion ? baseVersion : undefined,
     current_version: currentVersion,
     target_version: targetVersion,
     merged_version: mergedVersion,
 
     diff_outcome: diffOutcome,
-    merge_outcome: mergeOutcome,
+    conflict,
     has_update: valueCanUpdate,
-    has_conflict: mergeOutcome === ThreeWayMergeOutcome.Conflict,
   };
 };
 
 interface MergeResult {
-  mergeOutcome: ThreeWayMergeOutcome;
   mergedVersion: string;
+  conflict: ThreeWayDiffConflict;
 }
 
 interface MergeArgs {
+  hasBaseVersion: boolean;
   baseVersion: string | MissingVersion;
   currentVersion: string;
   targetVersion: string;
   diffOutcome: ThreeWayDiffOutcome;
 }
 
 const mergeVersions = ({
+  hasBaseVersion,
   baseVersion,
   currentVersion,
   targetVersion,
   diffOutcome,
 }: MergeArgs): MergeResult => {
   switch (diffOutcome) {
-    case ThreeWayDiffOutcome.StockValueNoUpdate:
-    case ThreeWayDiffOutcome.CustomizedValueNoUpdate:
-    case ThreeWayDiffOutcome.CustomizedValueSameUpdate: {
+    case ThreeWayDiffOutcome.StockValueNoUpdate: // Scenarios AAA and -AA
+    case ThreeWayDiffOutcome.CustomizedValueNoUpdate: // Scenario ABA
+    case ThreeWayDiffOutcome.CustomizedValueSameUpdate: // Scenario ABB
```

Comment:
Nice idea to put these scenarios in here, I think I'll incorporate this into some other diff files as well

Reply:
@dplumlee I ended up removing these, since now each scenario has its own `ThreeWayDiffOutcome` and there's no case like `// Scenarios AAA and -AA` anymore.